doc/0003-wizarding-api.md

```diff
@@ -0,0 +1,227 @@
+the given element and action.
+```ts
+type CanInspect: (tagName: string) => boolean
+type CanCreate: (tagName: string) => boolean
+```
+
+##### Requests
+
+We request a wizard through events.
+
+For viewing and/or editing purposes we can request an inspection.  
+The event is called `oscd-wizard-inspection-request` and its details look like this:
+```ts
+type WizardInspectionRequest = WizardRequest & {
+  element: Element,
+}
+```
+
+The other possible request is to create a new element under a parent element.  
+The event is called `oscd-wizard-creation-request` and its details look like this
+```ts
+type WizardCreationRequest = WizardRequest & {
+  parent: Element,
+  tagName?: string,
+}
+```
+
+Both event detail can contain an additional `stuff` for flexibility. `stuff` can be something
+that the wizard defines for additional configurations (e.g.: display it in readonly mode),
+or the editor can provide additional information about the element.
+
+> **⚠️ WARNING:** Please only use `stuff` if there is no other way.
+> It creates a tight coupling with a specific wizard or editor and it cannot be
+> guaranteed that your expectation about the `stuff` will be met!
+
+```ts
+type WizardRequest = {
+	stuff?: unknown
+}
+```
+
+> **ℹ️ INFO:** you can find the up-to-date definitions here: TODO: file link
+
+> **ℹ️ INFO**: More about other alternatives of requests: 
+> [Alternatives for Requests](#alternatives-of-requests)
+
+##### Finding Available Wizards
+
+When `Core` receives one of the above mentioned events it will go and ask the
+wizards if they can handle the element and action.
+It will call:
+- `CanInspect` if the event is `oscd-wizard-inspection-request` 
+- `CanCreate` if the event is `oscd-wizard-creation-request`.
+
+##### Initiation
+
+In case of inspection the wizard would get the element. 
+```html
+<oscd-wizard-type-templates .element={element} .stuff={stuff}></oscd-wizard-type-templates>
+```
+
+If we want to create a new element it will get the parent element and the new tag name it has to create. Something like this:
+```html
+<oscd-wizard-type-templates .parentElement={parentElement} tag-name="DO" .stuff={stuff}></oscd-wizard-type-templates>
+```
+
+
+##### Wizarding Process
+
+This part happens entirely in the wizard plugin. 
+It can modify the XML document or just display it in some user-friendly way.
+
+##### Self Closing
+
+The wizard decides when it is finished and closes itself. 
+
+`Core` has always the possibility to close the wizard. 
+For example when clicking on the backdrop of a dialog.
+
+----
+
+## Considered Options
+
+### Define handled tag names in the plugin configuration
+
+An alternative to the `CanHandle` static function is to define the
+tag names in the plugin description e.g: in a string array called `handledTags`.
+
+```ts
+type WizardPlugin = Plugin & {
+    handledTags: string[];
+}
+```
+
+The advantage is that we can enforce their definition,
+what we cannot do with a static functions as of now.
```

Comment:
What do you mean by "enforce their definition"? Who would do the enforcing, and how?

Reply:
TypeScript would, wouldn't it?
If you want to define a wizard plugin you would need to define ´handledTags´

Reply:
Plugins don't need to be written using TypeScript at all, I've written them in straght JS before, but a plugin could just as well be written in ClojureScript or Elm, for example.